Firmware/Board/v3/Src/gpio.c

```diff
@@ -213,8 +213,10 @@ bool GPIO_subscribe(GPIO_TypeDef* GPIO_port, uint16_t GPIO_pin,
   struct subscription_t* subscription = NULL;
   for (size_t i = 0; i < n_subscriptions; ++i) {
     if (subscriptions[i].GPIO_port == GPIO_port &&
-        subscriptions[i].GPIO_pin == GPIO_pin)
+        subscriptions[i].GPIO_pin == GPIO_pin) {
       subscription = &subscriptions[i];
+      break;
```

Comment:
Wouldn't that always give the _last_ match instead of first one? Is that good?
Also we don't really have very good timing determinism otherwise, and registering interrupts doesn't happen cyclically, so being fast is probably better.

Reply:
Fair point on the determinism, I left it like this now. The struct is zeroed out on unsubscribing so there should actually only ever be one matching item in the array.